sql/core/src/test/scala/org/apache/spark/sql/streaming/StreamingJoinSuite.scala

```diff
@@ -712,5 +709,32 @@ class StreamingOuterJoinSuite extends StreamTest with StateStoreMetricsTest with
       assertNumStateRows(total = 2, updated = 2)
     )
   }
-}
 
+  test("SPARK-29438: ensure UNION doesn't lead stream-stream join to use shifted partition IDs") {
+    withSQLConf(AUTO_BROADCASTJOIN_THRESHOLD.key -> "1") {
```

Comment:
Would we be better to add some comment here to explain why set AUTO_BROADCASTJOIN_THRESHOLD with "1", otherwise it maybe a little weird.

Reply:
I'll push a new test code which don't rely on broadcast join soon.

Reply:
It is much better. LGTM